G1ANT.Addon.Selenium/Api/SeleniumWrapper.cs

```
@@ -403,6 +404,50 @@ public string GetTextValue(SeleniumCommandArguments search, TimeSpan timeout)
             return res;
         }
 
+        public DataTable GetTableElement(SeleniumCommandArguments search, TimeSpan timeout)
+        {
+            PreCheckCurrentWindowHandle();
+            if (!string.IsNullOrEmpty(search.IFrameSearch?.Value))
+                webDriver.SwitchTo().Frame(FindElement(search.IFrameSearch.Value, search.IFrameBy.Value, timeout));
+            var element = FindElement(search.Search.Value, search.By.Value, timeout);
+
+            if (element.TagName != "table")
+            {
+                try
+                {
+                    element = element.FindElement(By.TagName("table"));
+                }
+                catch (Exception)
+                {
+                    throw new Exception("There are no tables found within the element specified. Try to change the search phrase");
+                }
+            }
```

it's not written in User Story, but I think we should throw an exception when the user search does not immediately return an HTML table element, why we looking nested element? Please remember that element could be also null

---

We should? Ok, I''ll change it